src/proto/h1/role.rs

```diff
@@ -1033,6 +1033,16 @@ impl Http1Transaction for Client {
             Version::HTTP_2 => {
                 debug!("request with HTTP2 version coerced to HTTP/1.1");
                 extend(dst, b"HTTP/1.1");
+
+                // http/2 can sometimes split cookie headers into multiple haeder
+                // values, which is not compatible with http/1.
+                //
+                // We re-join them on behalf of the client.
+                //
+                // https://tools.ietf.org/html/rfc7540#section-8.1.2.5
+                if let Some(joined_cookies) = unsplit_cookie_headers(&msg.head.headers) {
+                    msg.head.headers.insert(header::COOKIE, joined_cookies);
+                }
```

Comment:
Actually we probably just want to set a boolean value here, and use it later in `encode_headers` with a new `header::COOKIE` arm that would prevent writing the header name multiple times, joining them instead, to avoid looking up into the header map and mutating it.